renderer/src/pages/wallet/BalanceControl.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useState } from 'react'
+import Button from 'src/components/Button'
+import Text from 'src/components/Text'
+import { formatFilValue } from 'src/lib/utils'
+import { Wallet } from 'src/hooks/StationWallet'
+import CheckmarkIcon from 'src/assets/img/icons/checkmark.svg?react'
+
+const BalanceControl = ({
+  walletBalance = '',
+  sendThreshold,
+  processingTransaction,
+  transfer
+}: {
+    walletBalance?: string;
+    sendThreshold: number;
+    processingTransaction: Wallet['processingTransaction'];
+    transfer: () => void;
+}) => {
+  const [status, setStatus] = useState<'accruing' | 'idle' | 'processing' | 'complete'>('idle')
+
+  const setBalanceStateAfterComplete = useCallback((oldStatus: typeof status, newStatus: typeof status) => {
+    if (oldStatus === 'processing') {
+      setStatus('complete')
+      setTimeout(() => setStatus(newStatus), 2000)
+    } else {
+      setStatus(newStatus)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    const hasSufficientBalance = Number(walletBalance) >= sendThreshold
+
+    if (processingTransaction) {
+      setStatus('processing')
+    } else if (!processingTransaction && hasSufficientBalance) {
+      setBalanceStateAfterComplete(status, 'idle')
+    } else if (!processingTransaction && !hasSufficientBalance) {
+      setBalanceStateAfterComplete(status, 'accruing')
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This useEffect sets `status`, and that should only change if:
 - there's a change in `processingTransaction`
 - there's a change in wallet balance

If I added `status` to the deps array, it would run in a loop. The linter complains because I use the `status` value inside the `useEffect`, but it shouldn't be a dependency in this case

Reply:
Interesting, if it works I'm happy 👍